torchmetrics/metric.py

```diff
@@ -99,17 +110,30 @@ def __init__(
         self._device = torch.device("cpu")
 
         if compute_on_step is not None:
-            warnings.warn("Argument `decimal_places` is deprecated in v0.8 and will be removed in v0.9")
+            warnings.warn(
+                "Argument `compute_on_step` is deprecated in v0.8 and will be removed in v0.9", DeprecationWarning
+            )
 
         self.dist_sync_on_step = dist_sync_on_step
         self.process_group = process_group
         self.dist_sync_fn = dist_sync_fn
         self._to_sync = True
         self._should_unsync = True
 
-        self._update_signature = inspect.signature(self.update)
-        self.update: Callable = self._wrap_update(self.update)  # type: ignore
-        self.compute: Callable = self._wrap_compute(self.compute)  # type: ignore
+        if is_overridden("update", self, Metric):
+            raise NotImplementedError(
```

Comment:
@Borda what is the correct error to raise here?

Reply:
How about `NameError`?

Reply:
I ended up changing it to `DeprecationWarning` as the user will already be forced to implement `_update` and `_compute` since they are abstract methods